File: app/src/main/java/com/lumiyaviewer/lumiya/render/shaders/AvatarProgram.java
package com.lumiyaviewer.lumiya.render.shaders;

import android.opengl.GLES20;

public class AvatarProgram extends BasicPrimProgram {
    public int uJointMap;
    public int uJointMapLength;
    public int uJointMatrix;
    public int uUseWeight;
    public int vWeight;

    public AvatarProgram() {
        super(Shader.AvatarVertexShader, Shader.PrimFragmentShader);
    }

    @Override
    protected void bindVariables() {
        super.bindVariables();
        this.vWeight = GLES20.glGetAttribLocation(this.handle, "vWeight");
        this.uJointMatrix = GLES20.glGetUniformLocation(this.handle, "uJointMatrix");
        this.uJointMap = GLES20.glGetUniformLocation(this.handle, "uJointMap");
        this.uJointMapLength = GLES20.glGetUniformLocation(this.handle, "uJointMapLength");
        this.uUseWeight = GLES20.glGetUniformLocation(this.handle, "uUseWeight");
    }
}

File: app/src/main/java/com/lumiyaviewer/lumiya/render/shaders/AvatarProgram.kt
package com.lumiyaviewer.lumiya.render.shaders

import android.opengl.GLES20

class AvatarProgram : BasicPrimProgram(Shader.AvatarVertexShader, Shader.PrimFragmentShader) {
    @JvmField var uJointMap = 0
    @JvmField var uJointMapLength = 0
    @JvmField var uJointMatrix = 0
    @JvmField var uUseWeight = 0
    @JvmField var vWeight = 0

    override fun bindVariables() {
        super.bindVariables()
        vWeight = GLES20.glGetAttribLocation(handle, "vWeight")
        uJointMatrix = GLES20.glGetUniformLocation(handle, "uJointMatrix")
        uJointMap = GLES20.glGetUniformLocation(handle, "uJointMap")
        uJointMapLength = GLES20.glGetUniformLocation(handle, "uJointMapLength")
        uUseWeight = GLES20.glGetUniformLocation(handle, "uUseWeight")
    }
}
